src/ops.rs

```diff
@@ -883,10 +883,54 @@ impl AesEncryptable for Message {
     }
 }
 
+impl UpdateFinalize for Message {
+    /// Returns nothing and simply appends the write data into self.data
+    ///
+    fn update(&mut self, write_data: &[u8]) {
+        self.msg.append(&mut write_data.to_owned());
+    }
+    /// Internally, this calls cshake, then
+    /// passes self.data and returns the result
+    fn finalize(self) -> Result<Vec<u8>, OperationError> {
+        if let Some(d) = self.d {
+            let value = d as u64;
```

Comment:
Im imagining how we are going to use this in the sliding window, so I think what we want to do now that we have this framework is to make the following changes:

- [ ] ```finalize``` should accept a single parameter called ```output_length: u64```
- [ ] get rid of ```value``` here and pass in ```output_length``` to cshake instead, this way we can control the blocksize output
- [ ]  set ```self.msg``` = the result of cshake here, and then return self.msg

## 2 things are happening here:

1. this allows us to use the output of this hasher as a block of keystream that we can encrypt or decrypt into the message
2. we can effectively control the size of the output block to encrypt a message of any size

Reply:
Got it.